h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -414,6 +418,8 @@ public void compute2() {
         if (rs != null) {
           if (rs.isReady()) {
             if (_computeHisto && !rs.hasHisto()) { // a.2 => compute rollups
+              final Vec vec = DKV.getGet(_vecKey);
```

Comment:
I think you are right here. The check should be removed because if we have "ready" RollupStats and there is a simultaneous delete() operation the rollups could only be installed before the Vec is marked as mutating.

Good catch, thank you!